File: Programmers/구명보트.cpp
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

int solution(vector<int> p, int l) {
    int answer = 0, idx = 0;
    sort(p.begin(), p.end());
    while(p.size() > idx) {
        int back = p.back(); p.pop_back();
        if(p[idx] + back <= l) {
            answer++;
            idx++;
        }
        else 
            answer++;
    }
    return answer;
}

File: Programmers/구명보트.java
import java.util.*;
class Solution {
	public int solution(int[] people, int limit) {
		int answer = 0;
		Arrays.sort(people);

		for (int i = 0, j = people.length - 1; i <= j; j--) {
			if(people[i] + people[j] <= limit) {
				answer++;
				i++;
			} else
				answer++;
				
		}
		return answer;
	}
}